src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -72,28 +72,37 @@ private RankerQuery(List<Query> queries, FeatureSet features, LtrRanker ranker)
      * @return the lucene query
      */
     public static RankerQuery build(PrebuiltLtrModel model) {
-        return build(model.ranker(), model.featureSet(), null, Collections.emptyMap());
+        return build(model.ranker(), model.featureSet(), new LtrQueryContext(null, Collections.emptySet()), Collections.emptyMap());
     }
 
     /**
      * Build a RankerQuery.
      *
-     * @param model The model
+     * @param model   The model
      * @param context the context used to parse features into lucene queries
-     * @param params the query params
+     * @param params  the query params
      * @return the lucene query
      */
-    public static RankerQuery build(LtrModel model, QueryShardContext context, Map<String, Object> params) {
+    public static RankerQuery build(LtrModel model, LtrQueryContext context, Map<String, Object> params) {
         return build(model.ranker(), model.featureSet(), context, params);
     }
 
-    private static RankerQuery build(LtrRanker ranker, FeatureSet features, QueryShardContext context, Map<String, Object> params) {
+    private static RankerQuery build(LtrRanker ranker, FeatureSet features, LtrQueryContext context, Map<String, Object> params) {
+        validateActiveFeatures(features, context);
         List<Query> queries = features.toQueries(context, params);
         return new RankerQuery(queries, features, ranker);
     }
 
+    private static void validateActiveFeatures(FeatureSet features, LtrQueryContext context) {
+        for (String featureName : context.getActiveFeatures()) {
+            if (!features.hasFeature(featureName)) {
+                throw new IllegalArgumentException("Feature : [" + featureName + "] " + "does not exist");
```

Comment:
nitpick: it feels a bit late to do this here, I mean it's perhaps preferable to do the check earlier so that you can tell the user that the missing feature is part of the `active_features` param in the sltr query.
Personally I'd move this check in `StoredLtrQueryBuilder#doToQuery` method.

Reply:
done